src/components/App.js

```diff
@@ -92,6 +149,119 @@ class App extends Component {
     }
   // end constructor
   }
+  // Gets property name when entered
+
+  // Gets value of the property textbox
+  getPropertyName(e) {
+    this.setState({property: e.target.value});
+    console.log('this.property', this.state.property);
+  }
+
+  // Clears the property textbox. Executed in saveProperty function
+  resetPropertyName() {
+    const propertyTextbox = document.getElementById('live-API-property-textbox');
+    propertyTextbox.value = '';
+    this.setState({property: undefined});
+    // target the id element
+    // reset its value
+    // reset this.property to undefined
+  }
+  // In Step 1, click the save button to add property to this.text object
+  // and clear the textbox
+  saveProperty(property) {
+    console.log('hi');
+    console.log('property', property);
+    if (!property) return;
+    this.text[property] = this.state.propertyArray;
+    console.log('this.text', this.text);
+    this.resetPropertyName();
+  }
+
+
+  /*
+  Experiencing Problems Here
+  */
+
+  componentDidMount() {
+    const Application = this;
+
+    // Event listener to prevent population of new highlight div on existing highlight div
+    $(document).on('click','*', function(){
+        return false;
+    });
+
+    $(document).on('click', '.liveAPI-highlight', function(e) {
+      e.stopImmediatePropagation();
+    });
+
+    $(document).on('click', '.liveAPI-highlight-wrapper', function(e) {
+      e.stopImmediatePropagation();
+    });
+
+    $(document).on('click', '.liveAPI-highlight-button', function(e) {
```

Comment:
A quick comment on the purpose of each function would be useful when reviewing and working on the code in the future.